bolt-simple-taproot.md

```diff
@@ -0,0 +1,994 @@
+2. types:
+    1. type: 2 (`verification_musig2_pubnonce`)
+    2: data:
+        * [`66*byte`:`nonces`]
+    1. type: 4 (`signing_musig2_pubnonce`)
+    2: data:
+        * [`66*byte`:`nonces`]
+
+##### Requirements
+
+The sender:
+
+ - MUST set `verification_musig2_pubnonce` and `signing_musig2_pubnonce` to the
+   `musig2` public nonce as specified by the `NonceGen` algorithm of
+   `bip-musig2`
+
+ - Aggregate the received local+remote public nonces (into two combined nonces)
+   using the `NonceAgg` algorithm of `bip-musig2`.
+
+The receiver:
+
+ - MUST reject the channel if `signing_musig2_pubnonce` or
+   `verification_musig2_pubnonce` are not present.
+
+#### `funding_created` Extensions
+
+The sender:
+
+  - MUST derive the aggregated `musig2` public key using the exchanged `funding_pubkey`s
+    - Before aggregating the two funding public keys, the keys MUST be sorted
+      using the `KeySort` routine of `bip-musig2`.
+
+  - MUST use the exchanged `signing_musig2_pubnonce` and
+    `verification_musig2_pubnonce` received from the remote party with the
+    `NonceAgg` algorithm to derive the combined nonce. 
+    - Local commitment combined nonce: `NonceAgg(verificationNonce, signingNonce)`
+        - Where `verificationNonce` is the local party's and `signingNonce` is
+          the remote party's.
+    - Remote Commitment combined nonce: `NonceAgg(signingNonce, verificationNonce)`
+        - Where `signingNonce` is the local party's and `verificationNonce` is
+          the remote party's.
+  - the `signature` field MUST be a `musig2` partial signature using the
+    aggregated key defined above (serialized according to the Partial Signature
+    Encoding section), and the two public nonces exchanged in prior messages.
+
+  - MUST generate the attached `signature` using the combined
+    Remote commitment combined nonce it generated in the prior step.
+
+The recipient:
+  - MUST validate that the 
+  - MUST validate the signature as a _partial_ signature based on the
+    `PartialSigVerifyInternal` (using the verification nonce of the local
+    party) algorithm of `bip-musig2`:
+
+    - if the partial signature is invalid, MUST fail the channel
+
+#### `funding_signed` Extensions
+
+The sender MUST set:
+
+  - `signature` to the valid signature, using its `funding_pubkey` for the
+    initial commitment transaction, as defined in this documnt.
+    - this signature MUST be a partial signature as defined by `bip-musig2`.
+
+The recipient:
+
+  - MUST validate the signature as a _partial_ signature based on the
+    `PartialSigVerifyInternal` algorithm of `bip-musig2`:
+
+    - if the partial signature is invalid, MUST fail the channel
+
+#### `funding_locked` Extensions
+
+We add a new TLV field to the `funding_locked` message:
+
+1. type: 36 (`funding_locked`)
+2. data:
+    * ...
+    * [`funding_locked_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `funding_locked_tlvs`
+2. types:
+    1. type: 0 (`verification_musig2_pubnonce`)
```

Comment:
should the types here remain 0 and 2, where in all other extensions, they're 2 and 4?

Reply:
As is the nonce values are static, so `4` is always used for the nonce type. `2` is always the type for partial sig with nonce, and `6` is always used for the partial signature type, and `8` is used for the shutdown nonces.